src/main/java/org/terasology/web/resources/worldMap/WorldMapResource.java

```diff
@@ -0,0 +1,232 @@
+        r /= imageSize;
+        r *= brightnessMultiplier;
+        g /= imageSize;
+        g *= brightnessMultiplier;
+        b /= imageSize;
+        b *= brightnessMultiplier;
+        // get rid of the alpha because it is always 255
+        return new Color(r, g, b).getRGB() & 0x00FFFFFF;
+    }
+
+    /**
+     * Get the y coordinate of the block at the given x and z coordinates where the block is at the surface.
+     * This method is heavily inspired by the renderCell method of the MinimapGrid class in the Minimap module.
+     * @param x the x coordinate of the block
+     * @param yEstimate an estimate of the y coordinate
+     * @param z the z coordinate of the block
+     * @return the y coordinate of the block
+     */
+    private int getSurfaceY(int x, int yEstimate, int z) {
+        final int yMinimum = 0;
+        final int yMaximum = 200;
+        int y = yEstimate;
+        Block block = worldProvider.getBlock(x, yEstimate, z);
+        if (isIgnoredByMap(block)) {
+            while (isIgnoredByMap(block)) {
+                --y;
+                block = worldProvider.getBlock(x, y, z);
+                if (y <= yMinimum) {
+                    return BLOCK_Y_DEFAULT;
+                }
+            }
+        } else {
+            while (!isIgnoredByMap(block)) {
+                ++y;
+                block = worldProvider.getBlock(x, y, z);
+                if (y >= yMaximum) {
+                    return BLOCK_Y_DEFAULT;
+                }
+            }
+            --y;
+        }
+        return y;
+    }
+
+    /**
+     * Determine if the given block should be ignored when trying to find the y-coordinate of a block.
+     * @param block the block to check.
+     * @return whether or not the block should be ignored.
+     */
+    private static boolean isIgnoredByMap(Block block) {
+        return block.isPenetrable() && !block.isWater() && !block.getURI().toString().equals("engine:unloaded");
+    }
+
+    /**
+     * load all chunks around the designated map area so that the texture of each can be obtained.
+     * @param center the location of the block in the center of the map.
+     * @param mapBlockWidth the width of the map.
+     * @param mapBlockLength the length of the map.
+     */
+    private void loadChunks(Vector3i center, int mapBlockWidth, int mapBlockLength) {
+        final int maximumVerticalChunks = 8;
+        LocationComponent locationComponent = new LocationComponent();
+        locationComponent.setWorldPosition(center.toVector3f());
+        RelevanceRegionComponent relevanceRegionComponent = new RelevanceRegionComponent();
+        relevanceRegionComponent.distance = new Vector3i(((int) Math.ceil((double) mapBlockWidth / ChunkConstants.SIZE_X) * 2) + 2, maximumVerticalChunks,
+                ((int) Math.ceil((double) mapBlockLength / ChunkConstants.SIZE_Z) * 2) + 2);
+        entityManager.create(locationComponent, relevanceRegionComponent);
```

Comment:
In my understanding the entity created here is never removed, so I think this can leak memory which in the long run (after a lot of requests) could cause problems (waste of memory) (does it sond reasonable to you?). I think a solution could be not ignoring the result returned here and instead return in to the caller of `loadChunks`; then on line 87 we could save the `EntityRef` in a variable, and somehow destroy it when the chunks are no longer necessary (e.g. around line 110 I think). I think the proper way should be calling [`EntityRef.destroy()`](https://github.com/MovingBlocks/Terasology/blob/971c78fd322ce81e9d067b6cea0bc882a6584222/engine/src/main/java/org/terasology/entitySystem/entity/EntityRef.java#L57) but someone with better knowledge of the entity system may be able to confirm or provide more information.